api/src/main/java/org/openmrs/event/EventEngine.java

```
@@ -316,45 +401,16 @@ public void onMessage(Message message) {
 				conn.start();
 
 			} catch (JMSException e) {
-				// TODO Auto-generated catch block. Do something smarter here.
-				e.printStackTrace();
+				log.error("Exception occurred while subscribing", e);

```

Why are we swallowing the exception here (still) rather than re-throwing it as a runtime exception?  I want my system to start up as expected or not at all.

---

Yeah, I'm not sure this is the "right" behaviour, but I was trying to minimize the actual behaviour changes, so it just replaces the current behaviour of calling `printStackTrace()` (which goes to the `catalina.out` file or it's equivalent) with something actually using the logging system.